src/main/java/com/andyinthecloud/githubsfdeploy/controller/GitHubSalesforceDeployController.java

```diff
@@ -774,6 +820,36 @@ else if(folders.length>2)
 		}
 	}
 
+	private static void downloadRepoToPath(Path tempDir, ContentsServiceEx contentService, RepositoryId repoId, String ref) throws Exception {
```

Comment:
Methodology here is, given a specific path and repo target, download it as a Zip and save it into a given path.
This logic is generally the same, the difference being we nest the target path a bit deeper for sub modules because they "download" into a nested folder inside the project rather than the top of it.